apisix/balancer/chash.lua

```diff
@@ -55,6 +55,7 @@ end
 
 function _M.new(up_nodes, upstream)
     local str_null = str_char(0)
+    local last_server_index
```

Comment:
The `last_server_index` should be in ctx scope, not the upstream one. We should not share the same index across different requests to an upstream.